tools/rpkg/tests/testthat/test_relational.R

```diff
@@ -314,6 +312,234 @@ test_that("Symmetric difference returns the symmetric difference", {
     expect_equal(rel_df, expected_result)
 })
 
+test_that("rel aggregate with no groups but a sum over a column, sums the column", {
+   rel_a <- rel_from_df(con, data.frame(a=c(1, 2), b=c(3, 4)))
+   aggrs <- list(sum = expr_function("sum", list(expr_reference("a"))))
+   res <- rel_aggregate(rel_a, list(), aggrs)
+   rel_df <- rel_to_altrep(res)
+   expected_result <- data.frame(sum=c(3))
+   expect_equal(rel_df, expected_result)
+})
+
+test_that("rel aggregate with groups and aggregate function works", {
+   rel_a <- rel_from_df(con, data.frame(a=c(1, 2, 5, 5), b=c(3, 3, 4, 4)))
+   aggrs <- list(sum = expr_function("sum", list(expr_reference("a"))))
+   res <- rel_aggregate(rel_a, list(expr_reference("b")), aggrs)
+   rel_df <- rel_to_altrep(res)
+   expected_result <- data.frame(b=c(3, 4), sum=c(3, 10))
+   expect_equal(rel_df, expected_result)
+})
+
+test_that("Window sum expression function test works", {
+#   select j, i, sum(i) over (partition by j) from a order by 1,2
+    rel_a <- rel_from_df(con, data.frame(a=c(1:8),b=c(1, 1, 2, 2, 3, 3, 4, 4)))
+    sum_func <- expr_function("sum", list(expr_reference("a")))
+    aggrs <- expr_window(sum_func, partitions=list(expr_reference("b")))
+    expr_set_alias(aggrs, "window_result")
+    window_proj <- rel_project(rel_a, list(expr_reference("a"), aggrs))
+    order_over_window <- rapi_rel_order(window_proj, list(expr_reference("window_result")))
+    res <- rel_to_altrep(order_over_window)
+    expected_result <- data.frame(a=c(1:8), window_result=c(3, 3, 7, 7, 11, 11, 15, 15))
+    expect_equal(res, expected_result)
+})
+
+test_that("Window count function works", {
+#   select a, b, count(b) over (partition by a) from a order by a
+    rel_a <- rel_from_df(con, data.frame(a=c(1:8),b=c(1, 1, 2, 2, 3, 3, 4, 4)))
+    count_func <- expr_function("count", list(expr_reference("a")))
+    count <-  expr_window(count_func, partitions=list(expr_reference("b")))
+    expr_set_alias(count, "window_result")
+    window_proj <- rel_project(rel_a, list(count))
+    res <- rel_to_altrep(window_proj)
+    expected_result <- data.frame(window_result=c(2, 2, 2, 2, 2, 2, 2, 2))
+    expect_equal(res, expected_result)
+})
+
+test_that("Window avg function works", {
+#     select a, b, avg(b) over (partition by a) from a order by a
+    rel_a <- rel_from_df(con, data.frame(a=c(1:8),b=c(1, 1, 2, 2, 3, 3, 4, 4)))
+    avg_func <- expr_function("avg", list(expr_reference("a")))
+    avg_window <- expr_window(avg_func, partitions=list(expr_reference("b")))
+    expr_set_alias(avg_window, "window_result")
+    window_proj <- rel_project(rel_a, list(avg_window))
+    ordered <- rel_order(window_proj, list(expr_reference("window_result")))
+    res <- rel_to_altrep(ordered)
+    expected_result <- data.frame(window_result=c(1.5, 1.5, 3.5, 3.5, 5.5, 5.5, 7.5, 7.5))
+    expect_equal(res, expected_result)
+})
+
+test_that("Window sum with Partition, order, and window boundaries works", {
```

Comment:
Ah there is a little problem with conflating the ordering of the aggregate inputs and the ordering of the frame. We eventually want to [treat these independently](https://dl.acm.org/doi/10.1145/3514221.3526184.)

Reply:
The link provided doesn't work for me. Can you show me a test example, I can convert it into the relational syntax and see if the orders are still respected

Reply:
@hawkfish the hyperlink included a trailing period, which is why I couldn't access the paper, but just did a quick read. 

I'm still not quite sure what orderings are not treated independently. 
- The output of the window can be ordered using a separate `rel_order` relation
- The order of the aggregate can be defined by passing an order to the window_function (i.e `percent_rank` or `dense_rank`). This order is then passed to the window `order_by` attribute. So the logic here is a little bit confusing I admit. I can change before merging if necessary. 

I still don't understand how the ordering of the aggregate input and the ordering of the frame are not treated independently. If the aggregate input needs to be ordered so that a specific frame can be defined, can't the user first project the data using an `order by` to do this? I'm not really a window expert, so I'm probably missing something, but I'm having trouble coming up with a SQL statement where the order of a window aggregate input and the order of a frame are independent and the statement can't be split using a CTE or some other combination of Projection -> window. 

Or I don't understand exactly what you mean by ordering of the aggregate inputs and ordering of the frame, which is also entirely possible.

Reply:
Indeed we need to separate the kinds of sorting here. Maybe we need a separate expression class for aggregates so we can pass the order by? Alternatively we could have an optional parameter to the expression that sets the ordering if set and errors otherwise.

Reply:
After a talk with @lnkuiper I understand the importance of separating the kinds of sorting. 

For this project & PR however, I think it is out of scope as the relational API currently doesn't support aggregates with ordering. `SELECT STRING_AGG(x,',' ORDER BY x ASC), STRING_AGG(x,y ORDER BY x ASC) FROM strings` cannot be expressed in the relational api, and this isn't even a window function. In addition, this PR is for a project that doesn't need this functionality.

As Hannes has mentioned, the proposed changes would require more changes to either the `expr_function` or the `rel_aggregate` functions, which could break some upstream functionality that already exists.

Reply:
Sorry - my browser helpfully inserted the period!  Removed for posterity.